docs/api.yaml

```diff
@@ -0,0 +1,286 @@
+                phone_number:
+                  type: string
+                  pattern: '^\+[1-9]\d{1,14}$'
+                address:
+                  $ref: '#/components/schemas/Hash'
+      responses:
+        '200':
+          description: Registration successful
+          content:
+            application/json:
+              schema:
+                type: boolean
+  /transaction_history:
+    get:
+      summary: Retrieve transaction history
+      parameters:
+        - in: query
+          name: identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/Hash'
+        - in: query
+          name: pagination
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: List of transactions
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Transaction'
+  /get_balance:
+    get:
+      summary: Get the current usd balance of an account
+      parameters:
+        - in: query
+          name: identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/Hash'
+      responses:
+        '200':
+          description: Balance of the account.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/U256'
+  /get_limit:
+    get:
+      summary: Get the account spending limit
+      parameters:
+        - in: query
+          name: identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/Hash'
+      responses:
+        '200':
+          description: Account current spending limit
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/U256'
+  /get_current_expense:
+    get:
+      summary: Get the cumulated expensed amount on the 7 previous days 
+      parameters:
+        - in: query
+          name: identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/Hash'
+      responses:
+        '200':
+          description: Cumulated expensed amount on the 7 previous days
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/U256'
+  /get_whitelisted_contracts:
+    get:
+      summary: Get whitelisted contracts
+      parameters:
+        - in: query
+          name: identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/Hash'
+        - in: query
+          name: pagination
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: Successful retrieval of whitelisted contracts
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Hash'
```

Comment:
should it be a list of contracts or also have a description of what the contract is? If this value is displayed to the user it will look nicer with a brief description (e.g. Ekubo, Briq, etc).

Reply:
So currently the goal is to only transfer usdc but the comment still holds we'll display the name of the person we send/receive money to